src/main/java/ca/uhn/fhir/jpa/starter/MessagePlainProvider.java

```diff
@@ -0,0 +1,185 @@
+package ca.uhn.fhir.jpa.starter;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.dao.DaoRegistry;
+import ca.uhn.fhir.jpa.dao.IFhirResourceDao;
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.annotation.Operation;
+import ca.uhn.fhir.rest.annotation.OptionalParam;
+import ca.uhn.fhir.rest.annotation.ResourceParam;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.MessageHeader;
+import org.hl7.fhir.r4.model.MessageHeader.ResponseType;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ResourceType;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+public class MessagePlainProvider {
+  DaoRegistry daoRegistry;
+  FhirContext fhirContext;
+  public MessagePlainProvider(DaoRegistry _daoRegistry, FhirContext _fhirContext){
+    daoRegistry = _daoRegistry;
+    fhirContext = _fhirContext;
+  }
+  @Operation(name="$process-message", manualResponse=true)
+  public void processMessage(
+      @OptionalParam(name="async") String async,
+      @OptionalParam(name="response-url") String responseUrl,
+      @ResourceParam Bundle bundleR4,
+      HttpServletResponse theServletResponse
+  ) {
+    IParser parser = fhirContext.newJsonParser();
+
+    // validate the bundle
+    if (bundleR4.getType() != Bundle.BundleType.MESSAGE) {
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("Bundle is not of type 'message'");
+      respondWithResource(oo, theServletResponse, parser);
```

Comment:
Each of these `respondWithResource` calls should be followed by a `return;` so that processing doesn't continue down the rest of the method and throw an exception. Yes the response is already sent, so there's no impact to the user, but it would be nice not to clutter logs with unnecessary exceptions.